CMakeLists.txt

```diff
@@ -375,12 +375,26 @@ hpx_option(HPX_WITH_FULL_RPATH BOOL
   ${HPX_FULL_RPATH_DEFAULT} ADVANCED)
 
 ################################################################################
-# HPX Compute configuration
+# HPX CUDA Compute configuration
 ################################################################################
 hpx_option(HPX_WITH_CUDA BOOL
-  "Enable CUDA support (default: OFF)" OFF ADVANCED)
+  "Enable both CUDA compute+support modules (default: OFF)" OFF ADVANCED)
 hpx_option(HPX_WITH_CUDA_CLANG BOOL
   "Use clang to compile CUDA code (default: OFF)" OFF ADVANCED)
+hpx_option(HPX_WITH_CUDA_COMPUTE BOOL
+  "Enable CUDA compute module (default: OFF) - "
+  "note: enabling this also enables HPX_WITH_CUDA_SUPPORT" OFF ADVANCED)
+hpx_option(HPX_WITH_CUDA_SUPPORT BOOL
+  "Enable basic support for CUDA futures (default: OFF)" OFF ADVANCED)
+if(HPX_WITH_CUDA)
+  set(HPX_WITH_CUDA_COMPUTE ON CACHE BOOL "Enable full HPX compute capability with CUDA backend" FORCE)
+  set(HPX_WITH_CUDA_SUPPORT ON CACHE BOOL "Enable generation of HPX futures from CUDA invocations" FORCE)
+endif()
+if (HPX_WITH_CUDA_COMPUTE)
+  set(HPX_WITH_CUDA ON CACHE BOOL "Enable CUDA compute backend as well as CUDA/HPX futures" FORCE)
+else()
+  set(HPX_WITH_CUDA OFF CACHE BOOL "Disable HPX compute CUDA backend and support" FORCE)
```

Comment:
The description should say what happens when you turn the option on, and the description should be the on line 381, 394, and 396.

The same goes for the two other options.

Reply:
Thanks!